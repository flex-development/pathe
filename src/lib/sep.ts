/**
 * @file sep
 * @module pathe/lib/sep
 */

/**
 * Path segment separator.
 *
 * **Note**: Also known as "directory separator".
 */
const sep: string = '/'

export default sep
