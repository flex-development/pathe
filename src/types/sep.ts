/**
 * @file Type Aliases - Sep
 * @module pathe/types/Sep
 */

import type { PosixSep, WindowsSep } from '@flex-development/pathe'

/**
 * Union of path segment separators.
 *
 * @see {@linkcode PosixSep}
 * @see {@linkcode WindowsSep}
 */
type Sep = PosixSep | WindowsSep

export type { Sep as default }
