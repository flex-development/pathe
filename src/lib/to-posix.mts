/**
 * @file toPosix
 * @module pathe/lib/toPosix
 */

import validateURLString from '#internal/validate-url-string'
import sep from '#lib/sep'

export default toPosix

/**
 * Make separators in `path` POSIX-compliant.
 *
 * Supports encoded separators (e.g. `%5C`, `%5c`).
 *
 * @category
 *  utils
 *
 * @param {string} path
 *  The path to handle
 * @return {string}
 *  `path` with POSIX-compliant separators
 */
function toPosix(path: string): string

/**
 * Make separators in `url` POSIX-compliant.
 *
 * Supports encoded separators (e.g. `%5C`, `%5c`).
 *
 * @category
 *  utils
 *
 * @param {string} url
 *  The `URL` to handle
 * @return {URL}
 *  `url` with POSIX-compliant separators
 */
function toPosix(url: URL): URL

/**
 * Make separators in `value` POSIX-compliant.
 *
 * Supports encoded separators (e.g. `%5C`, `%5c`).
 *
 * @category
 *  utils
 *
 * @param {URL | string} value
 *  The {@linkcode URL} or path to handle
 * @return {URL | string}
 *  `value` with POSIX-compliant separators
 */
function toPosix(value: URL | string): URL | string {
  validateURLString(value, 'value')

  if (typeof value === 'string') {
    return value.replace(/\\/g, sep)
      .replace(/(?:%5C)/g, '%2F')
      .replace(/(?:%5c)/g, '%2f')
  }

  return value.href = toPosix(value.href), value
}
