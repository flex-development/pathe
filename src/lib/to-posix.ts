/**
 * @file toPosix
 * @module pathe/lib/toPosix
 */

import validateString from '#internal/validate-string'
import sep from '#lib/sep'

/**
 * Make separators in `path` POSIX-compliant.
 *
 * @category
 *  utils
 *
 * @param {string} path
 *  Path to handle
 * @return {string}
 *  `path` with POSIX-compliant separators
 */
function toPosix(path: string): string {
  validateString(path, 'path')
  return path.replace(/\\/g, sep)
}

export default toPosix
