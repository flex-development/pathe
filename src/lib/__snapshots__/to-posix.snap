// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators (".\\\\file") 1`] = `"./file"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\") 1`] = `"/"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\\\\\?\\\\UNC") 1`] = `"//?/UNC"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\\\\\?\\\\UNC\\\\server\\\\share") 1`] = `"//?/UNC/server/share"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\\\\\server two\\\\shared folder\\\\file path.zip") 1`] = `"//server two/shared folder/file path.zip"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\\\\\server\\\\share\\\\file_path") 1`] = `"//server/share/file_path"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\\\\\user\\\\admin$\\\\system32") 1`] = `"//user/admin$/system32"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("\\\\foo\\\\C:") 1`] = `"/foo/C:"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:") 1`] = `"C:"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:.") 1`] = `"C:."`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:\\\\") 1`] = `"C:/"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:\\\\Windows\\\\system32;C:\\\\Windows;C:\\\\Program Files\\\\node\\\\") 1`] = `"C:/Windows/system32;C:/Windows;C:/Program Files/node/"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:\\\\abc") 1`] = `"C:/abc"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:\\\\another_path\\\\DIR\\\\1\\\\2\\\\33\\\\\\\\index") 1`] = `"C:/another_path/DIR/1/2/33//index"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:\\\\path%5Cdir%5cindex.html") 1`] = `"C:/path%2Fdir%2findex.html"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:\\\\path\\\\dir\\\\index.html") 1`] = `"C:/path/dir/index.html"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("C:abc") 1`] = `"C:abc"`;

exports[`unit:lib/toPosix > path > should return \`path\` with posix-compliant separators ("another_path\\\\DIR with spaces\\\\1\\\\2\\\\33\\\\index") 1`] = `"another_path/DIR with spaces/1/2/33/index"`;

exports[`unit:lib/toPosix > url > should return \`url\` with posix-compliant separators 1`] = `
{
  "href": "file:///C:/path%2Fdir%2findex.html",
  "pathname": "/C:/path%2Fdir%2findex.html",
}
`;
